tests/forms/unit.py

```diff
@@ -139,3 +145,285 @@ def test_unit_state():
+    search_exact = "exact" in form.cleaned_data["soptions"]
+
+    # filter pootle path
+    if pootle_path:
+        qs = qs.filter(store__pootle_path__startswith=pootle_path)
+
+    unit_filter = form.cleaned_data["filter"]
+
+    # unit filter
+    if unit_filter in ["translated", "untranslated", "fuzzy"]:
+        qs = qs.filter(state=getattr(pootle_store.models, unit_filter.upper()))
+    elif unit_filter == 'incomplete':
+        qs = qs.filter(
+            Q(state=pootle_store.models.UNTRANSLATED)
+            | Q(state=pootle_store.models.FUZZY))
+    elif unit_filter == 'suggestions':
+        qs = qs.filter(suggestion__state=SuggestionStates.PENDING).distinct()
+    elif unit_filter == 'user-suggestions':
+        qs = qs.filter(
+            suggestion__state=SuggestionStates.PENDING,
+            suggestion__user=form.request_user)
+    elif unit_filter == 'user-suggestions-accepted':
+        qs = qs.filter(
+            suggestion__state=SuggestionStates.ACCEPTED,
+            suggestion__user=form.request_user)
+    elif unit_filter == 'user-suggestions-rejected':
+        qs = qs.filter(
+            suggestion__state=SuggestionStates.REJECTED,
+            suggestion__user=form.request_user)
+    elif unit_filter == 'user-submissions':
+        qs = qs.filter(
+            submission__submitter=form.request_user,
+            submission__type__in=SubmissionTypes.EDIT_TYPES)
+
+    elif unit_filter == 'user-submissions-overwritten':
+        qs = (
+            qs.filter(submission__submitter=form.request_user,
+                      submission__type__in=SubmissionTypes.EDIT_TYPES)
+              .exclude(submitted_by=form.request_user)
+              .distinct())
+
+    if search and sfields:
+        qs = UnitTextSearch(qs).search(search, sfields, search_exact)
+
+    if form.cleaned_data.get("modified_since"):
+        qs = qs.filter(mtime__gte=form.cleaned_data["modified_since"])
+
+    if form.cleaned_data["uids"]:
+        sort_by = form.cleaned_data.get("sort_by", None)
+        if sort_by == "priority":
+            qs = qs.distinct().order_by("priority")
+        elif sort_by == "submitted_on":
+            qs = qs.distinct().order_by("submitted_on")
+        else:
+            qs = qs.distinct().order_by("pk")
+
+        found = False
+        units = {}
+        uids = []
+        for unit in qs:
+            if unit.pk == form.cleaned_data["uids"][0]:
+                found = True
+            if found:
+                uids.append(unit.pk)
+                units[unit.pk] = unit
+    else:
+        uids = qs.distinct().values_list("pk", flat=True)
+        units = dict([(unit.pk, unit) for unit in qs.distinct()])
+
+    return uids, units
+
+
+def _test_form_filter(form, result):
+    uids, expected = _calculate_results(form)
+    unit_count = 0
+    for group in result["unitGroups"]:
+        for result_store, info in group.items():
+            for result_unit in info['units']:
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Statement seems to have no effect](https://www.codacy.com/app/dwayne/pootle/file/1704058323/issues/source?bid=1990595&fileBranchId=2994326#l234)